### PR TITLE
Add Firebase to the project

### PR DESCRIPTION
**Pull request by razanor**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,9 +13,11 @@
     "@chakra-ui/react": "^2.8.2",
     "@emotion/react": "^11.11.4",
     "@emotion/styled": "^11.11.5",
+    "firebase": "^10.12.2",
     "framer-motion": "^11.2.5",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-router-dom": "^6.23.1"
```
Comment (yugisu-flux) [inline package.json:20]: Thanks for adding Router as well 🙏

File: src/FirebaseDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    getUsers();
+  };
+
+  const deleteUser = async (id: string) => {
+    const userDoc = doc(db, 'users', id);
+    await deleteDoc(userDoc);
+    getUsers();
+  };
+
+  const getUsers = async () => {
```
Comment (yugisu-flux) [inline src/FirebaseDemo.tsx:31]: What do you think about adding some Firebase hooks library? (like `react-firebase-hooks`, if there isn't a better alternative)

Comment (razanor) [reply src/FirebaseDemo.tsx:31]: Yes, I can do that. Thanks for suggesting!

**Review by echelonka — APPROVED** [review]

**Review by yugisu-flux — APPROVED** [review]
LGTM!

**Pull request merged**